File: javascript/easy/fizzBuzz.js
/**
 * @param {number} n
 * @return {string[]}
 */

FIZZ = "Fizz", BUZZ = "Buzz", FIZZ_BUZZ = FIZZ + BUZZ;

var fizzBuzz = function(n) {
    const results = new Array();
    for(let i = 1; i <= n; i++) {
        if(isMod(i, 3) && isMod(i, 5)) {
            results.push(FIZZ_BUZZ);
        } else if(isMod(i, 3)) {
            results.push(FIZZ);
        } else if(isMod(i, 5)) {
            results.push(BUZZ);
        } else {
            results.push(i.toString());
        }
    }
    return results;
};

var isMod = function(i, mod) {
    return ((i % mod) == 0);
};

File: ruby/easy/fizz_buzz.rb
FIZZ, BUZZ, FIZZ_BUZZ = "Fizz", "Buzz", "FizzBuzz"
# @param {Integer} n
# @return {String[]}
def fizz_buzz(n)
    results = Array.new
    for i in 1..n
        if is_mod(i, 3) and is_mod(i, 5)
            results.push(FIZZ_BUZZ)
        elsif is_mod(i, 3)
            results.push(FIZZ)
        elsif is_mod(i, 5)
            results.push(BUZZ)
        else
            results.push(i.to_s)
        end
    end
    return results
end

def is_mod(i, mod)
    return ((i % mod) == 0);
end
